src/robotFactory.js

```diff
@@ -1,10 +1,76 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.chipVersion = chipVersion;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+
+  getInfo() {
+    return `Robot: ${this.name
+    }, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  };
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  };
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, chipVersion, coords,
+    maxLoadWeight, currentLoad = null) {
+    super(name, weight, chipVersion, coords);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (cargo.weight <= this.maxLoadWeight
+        && this.currentLoad === null
+    ) {
```

Comment:
I understand there might be eslint errors, but anyway, try to make it less weird, example:
```suggestion
    const isWeightAllowed = cargo.weight <= this.maxLoadWeight;
    
    if (isWeightAllowed && !this.currentLoad) {
```